Test/Unit/Controller/Cart/EmailTest.php

```diff
@@ -0,0 +1,166 @@
+use Magento\Framework\App\Action\Context;
+use Magento\Framework\App\RequestInterface;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Quote\Model\Quote;
+use PHPUnit\Framework\TestCase;
+
+class EmailTest extends TestCase
+{
+    /**
+     * @var Context
+     */
+    private $context;
+
+    /**
+     * @var CheckoutSession
+     */
+    private $checkoutSession;
+
+    /**
+     * @var CustomerSession
+     */
+    private $customerSession;
+
+    /**
+     * @var Bugsnag
+     */
+    private $bugsnag;
+
+    /**
+     * @var CartHelper
+     */
+    private $cartHelper;
+
+    /**
+     * @var Quote quote
+     */
+    private $quote;
+
+    /**
+     * @var RequestInterface request
+     */
+    private $request;
+
+    /**
+     * @var Email
+     */
+    private $currentMock;
+
+    public function setUp()
+    {
+        $this->initRequiredMocks();
+        $this->initCurrentMock();
+
+    }
+
+    public function testQuoteDoesNotExist()
+    {
+        $this->checkoutSession->method('getQuote')->willReturn(null);
+        $exception = new LocalizedException(__('Quote does not exist.'));
+        $this->bugsnag->expects($this->once())
+            ->method('notifyException')
+            ->with($this->equalTo($exception));
+        $this->currentMock->execute();
+    }
+
+    public function testQuoteIdFails()
+    {
+        $this->quote->method('getId')->willReturn(false);
+        $exception = new LocalizedException(__('Quote does not exist.'));
+        $this->bugsnag->expects($this->once())
+                    ->method('notifyException')
+                    ->with($this->equalTo($exception));
+        $this->currentMock->execute();
+    }
+
+    public function testNoEmail()
+    {
+        $this->checkoutSession->method('getQuote')->willReturn($this->quote);
+        $this->request->method('getParam')->willReturn('');
+        $exception = new LocalizedException(__('No email received.'));
+        $this->bugsnag->expects($this->once())
+                    ->method('notifyException')
+                    ->with($this->equalTo($exception));
+        $this->currentMock->execute();
+    }
+
+    /**
+     * This doesn't really seem to test invalid emails as the validateEmail method
+     * needs to be mocked. Generally just making sure that if validateEmail comes
+     * back false we error handle properly
+     */
+
+    public function testInvalidEmail()
+    {
+        $this->checkoutSession->method('getQuote')->willReturn($this->quote);
+        $this->request->method('getParam')->willReturn('invalidemail');
+        $this->cartHelper->method('validateEmail')->willReturn(false);
+        $exception = new LocalizedException(__('Invalid email: %1', 'invalidemail'));
+        $this->bugsnag->expects($this->once())
+                    ->method('notifyException')
+                    ->with($this->equalTo($exception));
+        $this->currentMock->execute();
+    }
+
+    public function testOtherException()
```

Comment:
Nit: Can you revisit some of these test names? A common pattern is to have the function being tested, and then what makes it special. For example, this would be: `testExecute_notifyException`.